build/mixxx.py

```diff
@@ -209,15 +209,15 @@ def __init__(self, target, machine, build, toolchain, available_features):
 
         # Now that environment variables have been read, we can detect the compiler.
         import subprocess
-        process = subprocess.Popen([self.env['CC'], '--version'], stdout=subprocess.PIPE)
+        process = subprocess.Popen("%s %s" %(self.env['CC'], '--version'), stdout=subprocess.PIPE, shell=True) # nosec
```

Comment:
Isn't shell=True redundant? The documentation of subprocess.Popen states that the command is executed through /bin/sh if the first argument is a string. I'm a Python novice.

Reply:
shell=True is required if you pass a full command as args[0]. It may also work without shell=False but that depends on the OS implementation.

I wonder why using the shell was required here. Can we clarify this with a comment.

Reply:
The way to use ccache is to export "ccache compiler" as CC and CXX. This does not work with the array notation of subprocess. You could shlib to process CC and then extract the argument array, but since this is a none security context i choose the easy way with shell=True.